tests/mypy_test.py

```diff
@@ -150,22 +151,22 @@ def parse_version(v_str: str) -> tuple[int, int]:
     return int(m.group(1)), int(m.group(2))
 
 
-def add_files(files: list[str], seen: set[str], root: str, name: str, args: TestConfig) -> None:
+def add_files(files: list[Path], seen: set[str], root: StrPath, name: str, args: TestConfig) -> None:
     """Add all files in package or module represented by 'name' located in 'root'."""
-    full = os.path.join(root, name)
-    mod, ext = os.path.splitext(name)
+    full = Path(root, name)
+    mod, ext = full.stem, full.suffix
     if ext in [".pyi", ".py"]:
         if match(full, args):
             seen.add(mod)
             files.append(full)
-    elif os.path.isfile(os.path.join(full, "__init__.pyi")) or os.path.isfile(os.path.join(full, "__init__.py")):
+    elif (full / "__init__.pyi").is_file() or (full / "__init__.py").is_file():
         for r, ds, fs in os.walk(full):
```

Comment:
Maybe we can replace this whole loop with something using `full.glob("**/*.{py,pyi}")`.

Reply:
I'm... actually not quite sure why we're collecting `.py` files here anyway. This part of `mypy_test.py` should probably only care about `.pyi` files!